### PR TITLE
Parse envelope json

### PR DESCRIPTION
**Pull request by rkondratowicz**
Will be used by the queue message processor in next PRs.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/EnvelopeParser.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.exceptions.InvalidMessageException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+
+import java.io.IOException;
+
+public class EnvelopeParser {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    public Envelope parse(byte[] bytes) {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/EnvelopeParser.java]: is it a candidate for static method ?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+public class Envelope {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: I think this class should ignore missing json fields for e.g caseReference

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/EnvelopeParserTest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+        // when
+        Throwable exc = catchThrowable(() -> envelopeParser.parse(json.getBytes()));
+
+        // then
+        assertThat(exc).isInstanceOf(InvalidMessageException.class);
+    }
+
+    @Test
+    public void should_throw_an_exception_if_json_envelope_has_fields_missing() throws Exception {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/EnvelopeParserTest.java]: Should not throw exception if case reference(also there may some others) is missing.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/EnvelopeParser.java (hunk 1 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/EnvelopeParser.java]: done

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    public Envelope(
+        @JsonProperty(value = "id", required = true) String id,
+        @JsonProperty(value = "case_ref", required = true) String caseRef,
+        @JsonProperty(value = "jurisdiction", required = true) String jurisdiction,
+        @JsonProperty(value = "doc_urls", required = true) List<String> docUrls
+    ) {
+        this.id = id;
+        this.caseRef = caseRef;
+        this.jurisdiction = jurisdiction;
+        this.docUrls = docUrls;
```
Comment (luigibk) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/model/Envelope.java]: If you really want it to be final you should make this an ImmutableList

**Pull request merged**